go-controller/pkg/ovn/egressfirewall_test.go

```diff
@@ -396,12 +424,149 @@ var _ = Describe("OVN test basic functions", func() {
 						Port:     400,
 					},
 				},
-				expectedMatch: "(udp && ( udp.dst == 400 )) || (tcp && ( tcp.dst == 100 || tcp.dst == 102 )) || (sctp && ( sctp.dst == 13 ))",
+				expectedMatch: "((udp && ( udp.dst == 400 )) || (tcp && ( tcp.dst == 100 || tcp.dst == 102 )) || (sctp && ( sctp.dst == 13 )))",
 			},
 		}
 		for _, test := range testcases {
 			l4Match := egressGetL4Match(test.ports)
 			Expect(test.expectedMatch).To(Equal(l4Match))
 		}
 	})
+	It("computes correct match function", func() {
+		type testcase struct {
+			ipv4source   string
+			ipv6source   string
+			destinations []matchTarget
+			ports        []egressfirewallapi.EgressFirewallPort
+			output       string
+		}
+		_, cidr, _ := net.ParseCIDR("10.128.0.0/14")
+		config.Default.ClusterSubnets[0].CIDR = cidr
+		testcases := []testcase{
+			{
+				ipv4source:   "testv4",
+				ipv6source:   "",
+				destinations: []matchTarget{{matchKindV4CIDR, "1.2.3.4/32"}},
+				ports:        nil,
+				output:       "match=\"(ip4.dst == 1.2.3.4/32) && ip4.src == $testv4 && ip4.dst != 10.128.0.0/14\"",
+			},
+			{
+				ipv4source:   "testv4",
+				ipv6source:   "testv6",
+				destinations: []matchTarget{{matchKindV4CIDR, "1.2.3.4/32"}},
+				ports:        nil,
+				output:       "match=\"(ip4.dst == 1.2.3.4/32) && (ip4.src == $testv4 || ip6.src == $testv6) && ip4.dst != 10.128.0.0/14\"",
+			},
+			{
+				ipv4source:   "testv4",
+				ipv6source:   "testv6",
+				destinations: []matchTarget{{matchKindV4AddressSet, "destv4"}, {matchKindV6AddressSet, "destv6"}},
+				ports:        nil,
+				output:       "match=\"(ip4.dst == $destv4 || ip6.dst == $destv6) && (ip4.src == $testv4 || ip6.src == $testv6) && ip4.dst != 10.128.0.0/14\"",
+			},
+			{
+				ipv4source:   "testv4",
+				ipv6source:   "",
+				destinations: []matchTarget{{matchKindV4AddressSet, "destv4"}, {matchKindV6AddressSet, ""}},
+				ports:        nil,
+				output:       "match=\"(ip4.dst == $destv4) && ip4.src == $testv4 && ip4.dst != 10.128.0.0/14\"",
+			},
+			{
+				ipv4source:   "testv4",
+				ipv6source:   "testv6",
+				destinations: []matchTarget{{matchKindV6CIDR, "2001::/64"}},
+				ports:        nil,
+				output:       "match=\"(ip6.dst == 2001::/64) && (ip4.src == $testv4 || ip6.src == $testv6) && ip4.dst != 10.128.0.0/14\"",
+			},
```

Comment:
question: does it make sense to add a testv6 only case?

Reply:
added